internal/highway/x/protocol/ipfs_lite.go

```diff
@@ -0,0 +1,203 @@
+package protocol
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"github.com/ipfs/go-cid"
+	"io/ioutil"
+	"sync"
+
+	"github.com/sonr-io/sonr/pkg/protocol"
+	"github.com/sonr-io/sonr/pkg/store"
+
+	ipfslite "github.com/hsanjuan/ipfs-lite"
+	"github.com/sonr-io/sonr/pkg/host"
+)
+
+var _ protocol.IPFS = (*IPFSLite)(nil)
+
+// NewIPFSLite creates a new IPFSLite instance with Host Implementation
+func NewIPFSLite(ctx context.Context, host host.SonrHost) (*IPFSLite, error) {
+	// Create IPFS Peer
+	ds := store.NewMemoryStore()
+	ipfsLite, err := ipfslite.New(ctx, ds.Batching(), host.Host(), host.Routing(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	p := &IPFSLite{
+		ctx:       ctx,
+		node:      host,
+		dataStore: ds,
+		peer:      ipfsLite,
+		lock:      sync.Mutex{},
+	}
+
+	p.peer.Bootstrap(ipfslite.DefaultBootstrapPeers())
+	return p, nil
+}
+
+// IPFSLite is an IPFS Lite implementation of ipfs.Protocol.
+type IPFSLite struct {
+	ctx       context.Context
+	node      host.SonrHost
+	dataStore *store.Memory
+	peer      *ipfslite.Peer
+	lock      sync.Mutex
+}
+
+func (i *IPFSLite) PinFile(ctx context.Context, cidstr string) error {
+	i.lock.Lock()
```

Comment:
i still think the mutex belongs in the memory store layer. W should move `i` to memory store and lock in that layer